crates/winrt-build/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    env, fs,
+    io::Write,
+    path::{Path, PathBuf},
+    process::Stdio,
+};
+
+pub struct Builder {
+    namespaces: Vec<String>,
+    output: PathBuf,
+    dependencies: Dependencies,
+}
+
+pub enum Dependencies {
+    Os,
+    Nuget(Vec<String>),
+    OsAndNuget(Vec<String>),
+}
+
+impl Default for Builder {
+    fn default() -> Self {
+        Self {
+            dependencies: Dependencies::Os,
+            namespaces: Vec::new(),
+            output: PathBuf::from(env::var("OUT_DIR").expect("No `OUT_DIR` env variable set"))
+                .join("bindings.rs"),
+        }
+    }
+}
+
+impl Builder {
+    pub fn insert_namespaces<T>(mut self, namespaces: T) -> Self
+    where
+        T: IntoIterator,
+        T::Item: std::string::ToString,
+    {
+        self.namespaces
+            .extend(namespaces.into_iter().map(|s| s.to_string()));
+        self
+    }
+
+    pub fn insert_nuget<T>(mut self, nuget_deps: T) -> Self
+    where
+        T: IntoIterator,
+        T::Item: std::string::ToString,
+    {
+        let new_deps = nuget_deps.into_iter().map(|i| i.to_string());
+        match &mut self.dependencies {
+            Dependencies::Os => self.dependencies = Dependencies::Nuget(new_deps.collect()),
+            Dependencies::Nuget(deps) => deps.extend(new_deps),
+            Dependencies::OsAndNuget(deps) => deps.extend(new_deps),
```

Comment:
It's not obvious how we can use this to get "OsAndNuget".

Reply:
Does it know that the SDK nuget package supersedes the "OS" metadata?

Reply:
You can't as of now. I can remove it or we can block this PR until I implement a way to do that.

Reply:
May need to wait since the nuget support doesn't yet support the sdk nuget.